server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java

```diff
@@ -712,6 +717,44 @@ public void run() {
           err.set(null);
           JOB_HOLDER.reset();
 
+          // consecutive failure processing
+          final long totalFailures =
+              errorHistory.values().stream().mapToLong(p -> p.getSecond().get()).sum();
+          if (totalFailures > 0) {
+            LOG.warn("This Compactor has had {} consecutive failures. Failures: {}", totalFailures,
+                errorHistory);
```

Comment:
Wondering how this will look, its always relogging the entire error history.  Also it seems like it will only log the first exception seen for a table, is that the intent?  I will know more about how this looks when I run some test.

Reply:
Yeah, I guess that I'm assuming the Throwable in `Map<TableId,Pair<Throwable,AtomicLong>>` would not really change in consecutive failures. I guess we could change it to `List<Pair<>>`, or remove the `Throwable` entirely and just make sure we log it. The exception class does get logged in the Coordinator when the compaction fails.

Reply:
This is what the logging looks like 

```
2025-07-15T17:46:35,339 [compactor.Compactor] WARN : This Compactor has had 1 consecutive failures. Failures: {1=(java.lang.ClassNotFoundException: org.apache.accumulo.testing.continuous.ValidatingIterator,1)}
```

Reply:
The summaries may be nice if you know to look for them.  Could do something like `Map<TableId, Map<ClassName, FailureCount>>`

Reply:
Ok, I missed your last comment before I posted mine. I can work on improving the error history map.

Reply:
Updated error history map in 3d71032